File: commands/rename.py
import discord
from discord.ext import commands
from Main import NoPermsEmbed, getPermissions

class Admin(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command()
    async def rename(self, ctx, newname):
        if getPermissions("channel"):
            oldname=ctx.channel.name
            embed=discord.Embed(
                title="Channel Renamed",
                description=f"From {oldname} to {newname}"
            )
            await ctx.channel.edit(name=newname)
            await ctx.send(embed=embed)
        else:
            await ctx.send(embed=NoPermsEmbed("Manage Channels"))


def setup(client):
    client.add_cog(Admin(client))

File: commands/rename.js
const { MessageEmbed } = require("discord.js");

exports.run = (client, message, args) => {
    const oldName = message.channel.name;
    const channel = message.channel;
    if (!message.guild.me.hasPermission("MANAGE_CHANNELS")) {
        return message.channel.send("You don't have the perms to do this!")
    }
    else {
        let newname = args.slice(0).join("-");
        const embed = new MessageEmbed()
            .setTitle("Channel Renamed")
            .setDescription(`Name Changed from ${oldName} to ${newname}`)
        channel.setName(newname)
            .then(channel.send(embed))
    }
}